seg_lapa/train.py

```diff
@@ -114,9 +116,9 @@ def main(cfg: DictConfig):
         print("\nResolved Dataclass:\n", config, "\n")
 
     wb_logger = config.logger.get_logger(cfg)
-    callbacks = config.callbacks.get_callbacks_list()
+    model = DeeplabV3plus(config, wb_logger.experiment.id)
+    callbacks = config.callbacks.get_callbacks_list(model.logs_dir)
```

Comment:
It feels weird to couple callbacks and LightningModule like this. 
Let's create a function in train.py that creates the log directory, and pass this directory to both the pl.LightningModule and the Callbacks:

We can also generate a run_id and use that to init wandb
```
    LOGS_DIR = 'logs'

    def create_log_dir(self, cfg):
        # Set the run ID: Read from config if resuming training, else generate unique id
        run_id = wandb.util.generate_id()
        resume_run = "never"  # Alternate: "allow"

        # Create log dir
        log_root_dir = get_project_root() / LOGS_DIR
        log_dir = log_root_dir / run_id
        log_dir.mkdir(parents=True, exist_ok=True)

        # Save the input config file to logs dir
        OmegaConf.save(cfg, log_dir / "train.yaml")
        
        return log_dir, run_id
```

You'll need to modify the WandbLogger:
1. Remove run_id from wandb config. Init wandb using our generated run id.
2. Set the save_dir of wandb to also be in project root

```
wb_logger = config.logger.get_logger(cfg, run_id)
```
```
@dataclass(frozen=True)
class WandbConf(LoggerConf):
    """Weights and Biases. Ref: wandb.com"""

    entity: str
    project: str
    run_name: Optional[str]

def get_logger(self, cfg: DictConfig, run_id: str, save_dir: str) -> pl_loggers.WandbLogger:
    wb_logger = pl_loggers.WandbLogger(name=run_name, id=run_id, save_dir=save_dir config=cfg_dict, **args_dict)

```